File: packages/ui-flutter/lib/src/hover_card.dart
import 'package:flutter/widgets.dart';

import 'popover.dart';

/// Mirrors `packages/ui/src/components/hover-card.tsx`. On the web it
/// opens on hover; here it's the same anchored overlay as [KinetixPopover]
/// with a caller-driven `visible` (wire it to a `MouseRegion` on
/// desktop). The other ports likewise built HoverCard on their Popover.
class KinetixHoverCard extends StatelessWidget {
  const KinetixHoverCard({
    super.key,
    required this.visible,
    required this.onDismiss,
    required this.anchor,
    required this.child,
  });

  final bool visible;
  final VoidCallback onDismiss;
  final Widget anchor;
  final Widget child;

  @override
  Widget build(BuildContext context) {
    return KinetixPopover(
      visible: visible,
      onDismiss: onDismiss,
      anchor: anchor,
      child: child,
    );
  }
}

File: packages/ui-swiftui/Sources/KinetixUI/HoverCard.swift
//
// HoverCard.swift — KinetixHoverCard.
//
// Mirrors packages/ui/src/components/hover-card.tsx. On the web it opens
// on hover; here it's the same anchored overlay as KinetixPopover with a
// caller-driven `isPresented` (wire it to `.onHover` on macOS) — the
// Compose port likewise built HoverCard on its Popover.
//

import SwiftUI

public struct KinetixHoverCard<Anchor: View, Content: View>: View {
    @Binding private var isPresented: Bool
    private let anchor: Anchor
    private let cardContent: Content

    public init(
        isPresented: Binding<Bool>,
        @ViewBuilder anchor: () -> Anchor,
        @ViewBuilder content: () -> Content
    ) {
        self._isPresented = isPresented
        self.anchor = anchor()
        self.cardContent = content()
    }

    public var body: some View {
        KinetixPopover(isPresented: $isPresented) {
            anchor
        } content: {
            cardContent
        }
    }
}
